src/brevitas/graph/equalize.py

```
@@ -1613,6 +1615,10 @@ def _compute_rotations(
                     logging.info("Skipping region")
                     continue
 
+        hidden_dim = hidden_dim if not region.expand_region else expanded_hidden_dim
+        if block_rotation_dim is not None and hidden_dim // block_rotation_dim > 1 and hidden_dim % block_rotation_dim == 0:
```

I would raise an warning in case `hidden_dim` is not divisible by `block_rotation_dim`  (or even an error?), so it does not silently ignore the passed value.

---

Warning, not an error, just because there is no way of deciding where to apply block hadamard so I would rather have a fallback